src/utils/mergeProps.test.ts

```diff
@@ -0,0 +1,56 @@
+import { mergeProps } from "./mergeProps";
+
+interface Props {
+  prop1?: string;
+  prop2?: string;
+  prop3?: string | null;
+}
+
+describe("mergeProps", () => {
+  it("it should join objects", () => {
+    const props: Props = {
+      prop1: "test",
+    };
+
+    const defaultProps = {
+      prop2: "test",
+    } as const;
+
+    expect(mergeProps(props, defaultProps)).toMatchObject({
```

Comment:
isn't `.toStrictEqual` better?
with `.toMatchObject` extra keys won't fail you.